firedrake/plot.py

```diff
@@ -217,6 +232,17 @@ def plot(function_or_mesh,
         return _plot_mult(functions, num_sample_points, axes=axes, **kwargs)
     # Single Function...
     function = function_or_mesh
+    functions = prepare_functions(function, real)
+    if not real:
+        plots = []
+        for function in functions:
+            p = plot(function,  num_sample_points, axes=axes, plot3d=plot3d, real=True, **kwargs)
+            p.set_title(function.name())
+            plots.append(p)
+        return plots
+    else:
+        function = functions[0]
```

Comment:
Better to restructure like this: `function, = functions`. That way we get an error if this list ever has more than one entry.